lib/cadet_web/controllers/sourcecast_controller.ex

```diff
@@ -0,0 +1,101 @@
+defmodule CadetWeb.SourcecastController do
+  use CadetWeb, :controller
+  use PhoenixSwagger
+
+  alias Cadet.{Repo, Course}
+  alias Cadet.Course.{Sourcecast}
+
+  def index(conn, _params) do
+    sourcecasts = Sourcecast |> Repo.all() |> Repo.preload(:uploader)
+    render(conn, "index.json", sourcecasts: sourcecasts)
+  end
+
+  def create(conn, %{"sourcecast" => sourcecast}) do
+    result = Course.upload_sourcecast_file(conn.assigns.current_user, sourcecast)
+
+    case result do
+      {:ok, _nil} ->
+        send_resp(conn, 200, "OK")
+
+      {:error, {status, message}} ->
+        conn
+        |> put_status(status)
+        |> text(message)
+    end
+  end
+
+  def create(conn, _params) do
+    send_resp(conn, :bad_request, "Missing or invalid parameter(s)")
+  end
+
+  def delete(conn, %{"id" => id}) do
+    result = Course.delete_sourcecast_file(conn.assigns.current_user, id)
+
+    case result do
+      {:ok, _nil} ->
+        send_resp(conn, 200, "OK")
+
+      {:error, {status, message}} ->
+        conn
+        |> put_status(status)
+        |> text(message)
+    end
+  end
+
+  swagger_path :index do
+    get("/sourcecast")
+    description("Lists all sourcecast files")
+    produces("application/json")
+    summary("Shows all files")
```

Comment:
Can we separate this action such that we do not send every audio file every time the frontend requests for this resource.

What should happen is that the `index` sends a list of sourcecast information, and maybe a `fetch` action for retrieving a specific sourcecast, along with the associated audio file.

Reply:
nvm i was under the impression that the audio file itself was sent